pkg/apis/pipeline/v1beta1/pipelinerun_types.go

```diff
@@ -200,6 +210,32 @@ type PipelineRunStatus struct {
 	PipelineRunStatusFields `json:",inline"`
 }
 
+// PipelineRunSucceededReason represents a reason for the pipeline run "Succeeded" condition
+type PipelineRunSucceededReason string
```

Comment:
nit: I feel this is coming out too long, can we rename this `PipelineRunSucceededReason` to `PipelineRunReason`, drop `Succeeded`? I understand it represents the `Succeeded` condition which could have status as failure or success depending on the execution but it could be implicit and just call it `PipelineRunReason`. Similar comment to all below reasons 😲  Rename `PipelineRunSucceededReasonStarted` to `PipelineRunReasonStarted` ...

Reply:
Yeah, I was debating about this one too. 
I thought we might have more conditions in future, and being on the API I wanted to have it unique... but no be fair there would be nothing wrong with having the same reason used in different conditions, so I'll go ahead and shorten the names :)